js/src/views/Web/web.js

```diff
@@ -0,0 +1,209 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+
+import styles from './web.css';
+
+export default class Web extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  }
+
+  state = {
+    token: null,
+    isLoading: true,
+    displayedUrl: this.lastAddress(),
+    url: this.lastAddress()
+  };
+
+  handleUpdateUrl = (url) => {
+    try {
+      window.localStorage.setItem('parity-web-lastAddress', url);
+    } catch (e) {}
+
+    this.setState({
+      isLoading: true,
+      displayedUrl: url,
+      url: url
+    });
+  };
+
+  handleOnRefresh = (ev) => {
+    // TODO [ToDr]
+    this.setState({
+      isLoading: true,
+      url: `${this.state.displayedUrl}?t=${Date.now()}`
+    });
+  };
+
+  handleIframeLoad = (ev) => {
+    this.setState({
+      isLoading: false
+    });
+  };
+
+  lastAddress () {
+    let res = null;
+    try {
+      res = window.localStorage.getItem('parity-web-lastAddress');
+    } catch (e) {}
+    return res || 'https://mkr.market';
+  }
+
+  componentDidMount () {
+    this.context.api.signer.generateWebProxyAccessToken().then(token => {
+      this.setState({ token });
+    });
+  }
+
+  address () {
+    const { dappsUrl } = this.context.api;
+    const { url, token } = this.state;
+    const path = url.replace(/:/g, '').replace(/\/\//g, '/');
+
+    return `${dappsUrl}/web/${token}/${path}/`;
+  }
+
+  render () {
+    const { displayedUrl, isLoading, token } = this.state;
+    const address = this.address();
+
+    if (!token) {
+      return (
+        <div
+          className={ styles.wrapper }
+          >
+          <h1 className={ styles.loading }>
+            Requesting access token...
+          </h1>
+        </div>
+      );
+    }
+
+    return (
+      <div
+        className={ styles.wrapper }
+        >
+        <AddressBar
+          url={ displayedUrl }
+          isLoading={ isLoading }
+          onChange={ this.handleUpdateUrl }
+          onRefresh={ this.handleOnRefresh }
+        />
+        <iframe
+          className={ styles.frame }
+          frameBorder={ 0 }
+          name={ name }
+          sandbox='allow-forms allow-same-origin allow-scripts'
+          scrolling='auto'
+          onLoad={ this.handleIframeLoad }
+          src={ address } />
+      </div>
+    );
+  }
+}
+
+class AddressBar extends Component {
+
+  static propTypes = {
+    isLoading: PropTypes.bool.isRequired,
+    url: PropTypes.string.isRequired,
```

Comment:
Alphabetical order so nobody needs to wonder is preferred here with properties, state, passing props, imports, etc. (Just making some sense of the chaos)